File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.java
class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        int[] answer = {};
        
        int xMove = 0;
        int yMove = 0;
        
        int xLimit = (board[0]-1)/2;
        int yLimit = (board[1]-1)/2;
        
        for (String k: keyinput) {
            
            int newX = xMove;
            int newY = yMove;
            
            switch (k) {
                case "left":
                    newX -= 1;
                    break;
                case "right":
                    newX += 1;
                    break;
                case "up":
                    newY += 1;
                    break;
                case "down":
                    newY -= 1;
                    break;
            }
            
            if (newX >= -xLimit && newX <= xLimit) {
                xMove = newX;
            }
            if (newY >= -yLimit && newY <= yLimit) {
                yMove = newY;
            }
        }
        
        return new int[]{xMove, yMove};
    }
}

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):
    answer = []
    
    x_move = 0
    y_move = 0
    
    x_limit = (board[0]-1)//2
    y_limit = (board[1]-1)//2
    
    for k in keyinput:
        
        new_x, new_y = x_move, y_move
        
        match k:
            case "left":
                new_x -=1
            case "right":
                new_x +=1
            case "up":
                new_y +=1
            case "down":
                new_y -=1
        
        if -x_limit <= new_x <= x_limit:
            x_move = new_x
        if -y_limit <= new_y <= y_limit:
            y_move = new_y
    return [x_move, y_move]